### ISSUE TITLE
Gentoo ebuilds impossible if code gets downloaded at compile time. (v.0.18.0 and later)

### ISSUE THREAD

**Issue by misanthropos** [ja2-stracciatella/ja2-stracciatella#1377]
Emerging (download, compile and install) software on Gentoo runs in a sandbox with no internet access. Therefore no ebuilds can be made, since compilation will fail.

e.g.
```
[1/9] Creating directories for 'get-lua'

[2/9] Performing download step (download, verify and extract) for 'get-lua'
FAILED: get-lua-prefix/src/get-lua-stamp/get-lua-download 
```
The same is true for dependencies which download software at compile time (like lib-miniaudio).

Package maintainers should be able to set variables for software packages to indicate they are locally installed (like e.g. libsmacker). In that case a maintainer can write a gentoo-ebuild for that package. (like in 0.17.0).

An alternative could be to include all the sources needed in releases.

**Comment by lynxlynxlynx** [comment]
Isn't it all here?
https://github.com/ja2-stracciatella/ja2-stracciatella/tree/master/dependencies

But yeah, it looks like system lua is not supported yet.

**Comment by misanthropos** [comment]
If you look closely, lib-miniaudio is not there. I needs to be downloaded first.

**Comment by selaux** [comment]
[Here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/ja2-stracciatella/default.nix) is how I've done it for Nix for 0.17.0, which also runs in a no-network sandbox and requires you to download and verify all files beforehand.

Basically I ended up downloading the required dependencies separately and patching the respective CMake files to point to the separately downloaded file. I've no idea how Gentoo ebuilds work, but I guess something similar should be possible.

**Comment by misanthropos** [comment]
Sorry - I forgot to mention the version I was talking about: Its 0.18.0

The ebuild for 0.17.0 is up and part of gentoos repository. I have problems with the creation of an ebuild for 0.18.0

**Comment by misanthropos** [comment]
Great! Thank you for those changes.
However lib-miniaudio will still be downloaded at compile time.

**Comment by ja2-derek** [comment]
Maybe also same for `magic_enum` now.

**Comment by timp87** [comment]
Hi,
I'm currently maintaining this game port for FreeBSD, so I'm facing with the same problems as the @misanthropos 
The linked issue I did has fixed Lua and sol2.
The only thing left is miniaudio (at least in my case). It's a tricky one. Our FreeBSD port for miniaudio provides only a limited subset of files for this software, not enough for building ja2. I did an attempt to change the port and fix ja2 a year ago, but it was not successful.
For now I did a workaround in the ports tree that downloads the miniaudio and changes its path in ja2 files.
Maybe I'll look at this again soon.
